website/server/models/group.js

```diff
@@ -577,6 +578,19 @@ schema.statics.cleanGroupQuest = function cleanGroupQuest () {
   };
 };
 
+async function _updateUserWithRetries (userId, updates, numTry = 1) {
+  return await User.update({_id: userId}, updates).exec()
+    .then((raw) => {
```

Comment:
you should be able to use catch, it's used in multiple places with mongoose. Because we're plugging Bluebird in place of their promise implementation that doesn't support it